TYPO3.Flow/Classes/TYPO3/Flow/Mvc/Routing/RouterCachingService.php

```diff
@@ -49,6 +50,30 @@ class RouterCachingService
     protected $systemLogger;
 
     /**
+     * @Flow\Inject
+     * @var ObjectManagerInterface
+     */
+    protected $objectManager;
+
+    /**
+     * @Flow\InjectConfiguration("mvc.routes")
+     * @var array
+     */
+    protected $routingSettings;
+
+    /**
+     * @return void
+     */
+    public function initializeObject()
+    {
+        // flush routing caches if in Development context & routing settings changed
+        if ($this->objectManager->getContext()->isDevelopment() && $this->routeCache->get('routingSettings') !== $this->routingSettings) {
```

Comment:
This will only happen in Dev context upon initial request - but it will just flush an empty cache, which is not an issue as far as I can tell (and I just tested it as well)